### PR TITLE
Convert to new architecture

### PR DESCRIPTION
**Pull request by scorphus**

### CODE REVIEW

File: _python.fish [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Use python2 if found, otherwise fallback to python.
 function _python
-  if which python2
+  if available python2
```
Comment (derekstavis) [inline _python.fish:3]: Great one :+1:

File: pybeautifyjson.fish [hunk 2]
```diff
@@ -1,5 +1,7 @@
-# beautify json string
-# use : pybeautifyjson '{"foo": "lorem", "bar": "ipsum"}'
-function pybeautifyjson
-  echo $argv | _python -mjson.tool
+function pybeautifyjson -d "Beautify a JSON string"
```
Comment (derekstavis) [inline pybeautifyjson.fish:1]: Unrelated to this PR, in the future we could change this function to something smaller like `beautyjson`, which is 10 chars vs 14.

**Comment by derekstavis** [comment]
Awesome! LGTM! :+1:

File: pybeautifyjson.fish (hunk 2 above)
Comment (bobthecow) [inline pybeautifyjson.fish:1]: my local version of this (which is essentially identical, despite being independently invented ages ago) is called `jsonpp` for "json pretty print",

Comment (derekstavis) [inline pybeautifyjson.fish:1]: Much better! And allows to keep "namespacing" - `pyjsonpp`, which is an improvement over 10 chars.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+```fish
+$ omf install python
+```
+
+
+## Usage
+
+```fish
+$ pyclean  # recursively clean .pyc and .pyo
+$ pybeautifyjson '{"foo": "lorem", "bar": "ipsum"}'  # butify this JSON string
```
Comment (bpinto) [inline README.md]: `butify` -> `beautify`

Comment (derekstavis) [inline README.md]: eagle eye!

Comment (scorphus) [inline README.md]: Peer review FTW! Good catch, @bpinto, thanks!

**Comment by bpinto** [comment]
@scorphus go for it! :+1:

**Comment by derekstavis** [comment]
:+1:

**Comment by scorphus** [comment]
Thanks, fellas!

**Pull request merged**